apps/webcert/src/store/api/createAsyncApiThunk.ts

```diff
@@ -4,26 +4,29 @@ import axios, { isAxiosError } from 'axios'
 import type { RootState } from '../reducer'
 import { apiCallFailed } from './apiActions'
 import { createApiError, getHeaders } from './apiUtils'
-import { addRequest, isRequestLoading, removeRequest } from './requestSlice'
+import { addRequest, removeRequest, selectById } from './requestSlice'
 
 interface ApiCall {
   url: string
   method: 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH'
   data?: unknown
   headers?: Record<string, string>
+  functionDisablerType?: string
 }
 
 export function createAsyncApiThunk<Response, ThunkArg = void>(name: string, init: (args: ThunkArg) => ApiCall) {
+  const id = randomUUID()
+
   return createAsyncThunk<Response, ThunkArg, { state: RootState }>(name, async (args, { rejectWithValue, dispatch, getState }) => {
-    const id = randomUUID()
-    const { url, method, data, headers } = init(args)
+    const payload = init(args)
+    const { url, method, headers, data } = payload
 
-    if (isRequestLoading({ url, method })(getState())) {
-      return rejectWithValue(new Error('Request is already loading'))
+    if (selectById(getState(), id)) {
+      return rejectWithValue({ message: `[${method}] ${url} is pending`, payload })
```

Comment:
ID is now created only once for each `AsyncApiThunk`, this makes validating if an API request is currently pending much simpler.